### PR TITLE
WIP: Friend system

### PR DESCRIPTION
**Pull request by Elendan**

### CODE REVIEW

File: src/NosCore.Controllers/DefaultPacketController.cs [hunk 1]
```diff
@@ -330,6 +334,24 @@ public void WhisperPacket(WhisperPacket whisperPacket)
                 var messageData = whisperPacket.Message.Split(' ');
                 var receiverName = messageData[whisperPacket.Message.StartsWith("GM ") ? 1 : 0];
 
+                CharacterDTO target = DAOFactory.CharacterDAO.FirstOrDefault(s => s.Name == receiverName);
```
Comment (0Lucifer0) [inline src/NosCore.Controllers/DefaultPacketController.cs]: no! no database access

File: src/NosCore.Controllers/DefaultPacketController.cs [hunk 2]
```diff
@@ -404,5 +425,265 @@ public void WhisperPacket(WhisperPacket whisperPacket)
...
             }
         }
+
+        /// <summary>
+        ///     btk packet
+        /// </summary>
+        /// <param name="btkPacket"></param>
+        public void FriendTalk(BtkPacket btkPacket)
+        {
+            if (string.IsNullOrEmpty(btkPacket?.Message))
```
Comment (0Lucifer0) [inline src/NosCore.Controllers/DefaultPacketController.cs]: can't be null

File: src/NosCore.Controllers/DefaultPacketController.cs [hunk 3]
```diff
@@ -404,5 +425,265 @@ public void WhisperPacket(WhisperPacket whisperPacket)
...
+            Session.Character.DeleteBlackList(bldelPacket.CharacterId);
+        }
+
+        /// <summary>
+        ///     flPacket packet
+        /// </summary>
+        /// <param name="flPacket"></param>
+        public void AddDistantFriend(FlPacket flPacket)
+        {
+            if (string.IsNullOrEmpty(flPacket?.CharacterName))
```
Comment (0Lucifer0) [inline src/NosCore.Controllers/DefaultPacketController.cs]: cant be null

File: src/NosCore.Controllers/DefaultPacketController.cs [hunk 4]
```diff
@@ -404,5 +425,265 @@ public void WhisperPacket(WhisperPacket whisperPacket)
...
+            {
+                return;
+            }
+
+            CharacterDTO target = DAOFactory.CharacterDAO.FirstOrDefault(s => s.Name == flPacket.CharacterName);
```
Comment (0Lucifer0) [inline src/NosCore.Controllers/DefaultPacketController.cs]: no database access

File: src/NosCore.Controllers/DefaultPacketController.cs [hunk 5]
```diff
@@ -404,5 +425,265 @@ public void WhisperPacket(WhisperPacket whisperPacket)
...
+            AddFriend(fins);
+        }
+
+        /// <summary>
+        ///     blPacket packet
+        /// </summary>
+        /// <param name="blPacket"></param>
+        public void DistantBlackList(BlPacket blPacket)
+        {
+            if (string.IsNullOrEmpty(blPacket?.CharacterName))
```
Comment (0Lucifer0) [inline src/NosCore.Controllers/DefaultPacketController.cs]: cant be null

File: src/NosCore.Controllers/DefaultPacketController.cs [hunk 6]
```diff
@@ -404,5 +425,265 @@ public void WhisperPacket(WhisperPacket whisperPacket)
...
+            {
+                return;
+            }
+
+            CharacterDTO target = DAOFactory.CharacterDAO.FirstOrDefault(s => s.Name == blPacket.CharacterName);
```
Comment (0Lucifer0) [inline src/NosCore.Controllers/DefaultPacketController.cs]: no database access

File: src/NosCore.GameObject/Character.cs [hunk 7]
```diff
@@ -14,11 +20,32 @@ namespace NosCore.GameObject
...
         public bool IsChangingMapInstance { get; set; }
 
+        public ConcurrentDictionary<long, CharacterRelationDTO> CharacterRelations { get; set; }
+
+        public bool IsFriendListFull
+        {
+            get => _isFriendListFull;
+            set
+            {
+                _isFriendListFull = CharacterRelations.Where(s => s.Value.RelationType == CharacterRelationType.Friend).ToList().Count >= 80;
```
Comment (0Lucifer0) [inline src/NosCore.GameObject/Character.cs]: this need to be in the get and no set required

File: src/NosCore.GameObject/Character.cs [hunk 8]
```diff
@@ -87,6 +114,36 @@ const int
...
+
+            Parallel.ForEach(characterRelationPartitioner, relation =>
+            {
+                CharacterRelations[relation.CharacterRelationId] = relation;
+            });
+        }
+
+        public void RefreshRelations(long relationId)
+        {
+            CharacterRelationDTO relationDto = DAOFactory.CharacterRelationDAO.FirstOrDefault(s => s.CharacterRelationId == relationId);
```
Comment (0Lucifer0) [inline src/NosCore.GameObject/Character.cs]: No database access

File: src/NosCore.GameObject/Character.cs [hunk 9]
```diff
@@ -87,6 +114,36 @@ const int
 
         public byte Equipment { get; set; }
 
+        public void LoadRelations()
+        {
```
Comment (0Lucifer0) [inline src/NosCore.GameObject/Character.cs]: this doesnt require a method on character just put it on the loading for first connection on character

File: src/NosCore.GameObject/Character.cs [hunk 10]
```diff
@@ -162,6 +219,136 @@ public int IsReputHero()
...
+            var subpackets = new List<FinitSubPacket>();
+            foreach (CharacterRelationDTO relation in CharacterRelations.Values.Where(s => s.RelationType == CharacterRelationType.Friend || s.RelationType == CharacterRelationType.Spouse))
+            {
+                if (relation.RelatedCharacterId == CharacterId)
+                {
+                    continue;
+                }
+
+                string relatedCharacter = DAOFactory.CharacterDAO.FirstOrDefault(s => s.CharacterId == relation.RelatedCharacterId)?.Name;
```
Comment (0Lucifer0) [inline src/NosCore.GameObject/Character.cs]: nope

File: src/NosCore.GameObject/Character.cs [hunk 11]
```diff
@@ -162,6 +219,136 @@ public int IsReputHero()
...
+            if (relation == null)
+            {
+                Session.SendPacket(new InfoPacket
+                {
+                    Message = Language.Instance.GetMessageFromKey(LanguageKey.CANT_FIND_CHARACTER, Session.Account.Language)
+                });
+                return;
+            }
+
+            DAOFactory.CharacterRelationDAO.Delete(relation.CharacterRelationId);
```
Comment (0Lucifer0) [inline src/NosCore.GameObject/Character.cs]: nope

File: src/NosCore.GameObject/Character.cs [hunk 12]
```diff
@@ -162,6 +219,136 @@ public int IsReputHero()
...
+        public void AddRelation(long characterId, CharacterRelationType relationType)
+        {
+            var relation = new CharacterRelationDTO
+            {
+                CharacterId = CharacterId,
+                RelatedCharacterId = characterId,
+                RelationType = relationType
+            };
+
+            DAOFactory.CharacterRelationDAO.InsertOrUpdate(ref relation);
```
Comment (0Lucifer0) [inline src/NosCore.GameObject/Character.cs]: nope

File: src/NosCore.GameObject/Character.cs [hunk 13]
```diff
@@ -162,6 +219,136 @@ public int IsReputHero()
...
+            }
+
+            DAOFactory.CharacterRelationDAO.Delete(relation1.CharacterRelationId);
+            DAOFactory.CharacterRelationDAO.Delete(relation2.CharacterRelationId);
+            RefreshRelations(relation1.CharacterRelationId);
+            RefreshRelations(relation2.CharacterRelationId);
+            Session.SendPacket(GenerateFinit());
+            //TODO: Refresh target relation if online, this is just for testing purposes
+            ClientSession target = ServerManager.Instance.GetSessionByCharacterId(characterId);
+            target?.SendPacket(target?.Character.GenerateFinit());
```
Comment (0Lucifer0) [inline src/NosCore.GameObject/Character.cs]: you also need to remove from the relation

File: src/NosCore.GameObject/Networking/ServerManager.cs [hunk 14]
```diff
@@ -27,6 +32,48 @@ private ServerManager()
...
         public static ServerManager Instance => _instance ?? (_instance = new ServerManager());
 
+        public ClientSession GetSessionByCharacterId(long characterId)
+        {
+            return Sessions.FirstOrDefault(s => s.Value.Character.CharacterId == characterId).Value;
+        }
+
+        public CharacterDTO GetCharacterByCharacterId(long characterId)
+        {
+            return DAOFactory.CharacterDAO.FirstOrDefault(s => s.CharacterId == characterId);
```
Comment (0Lucifer0) [inline src/NosCore.GameObject/Networking/ServerManager.cs]: useless method

File: src/NosCore.GameObject/Networking/ServerManager.cs [hunk 15]
```diff
@@ -27,6 +32,48 @@ private ServerManager()
 
         public static ServerManager Instance => _instance ?? (_instance = new ServerManager());
 
+        public ClientSession GetSessionByCharacterId(long characterId)
+        {
+            return Sessions.FirstOrDefault(s => s.Value.Character.CharacterId == characterId).Value;
```
Comment (0Lucifer0) [inline src/NosCore.GameObject/Networking/ServerManager.cs]: useless methode stop adding things in servermanager

File: src/NosCore.GameObject/Networking/ServerManager.cs [hunk 16]
```diff
@@ -103,6 +149,27 @@ internal void UnregisterSession(ClientSession clientSession)
             Sessions.TryRemove(clientSession.SessionId, out _);
         }
 
+        public bool IsCharacterConnected(long characterId)
```
Comment (0Lucifer0) [inline src/NosCore.GameObject/Networking/ServerManager.cs]: shouldnt be in servermanager

File: src/NosCore.GameObject/Networking/ServerManager.cs [hunk 17]
```diff
@@ -27,6 +32,48 @@ private ServerManager()
...
+        {
+            return Sessions.FirstOrDefault(s => s.Value.Character.CharacterId == characterId).Value;
+        }
+
+        public CharacterDTO GetCharacterByCharacterId(long characterId)
+        {
+            return DAOFactory.CharacterDAO.FirstOrDefault(s => s.CharacterId == characterId);
+        }
+
+        public void UpdateFriendList(ClientSession session)
```
Comment (0Lucifer0) [inline src/NosCore.GameObject/Networking/ServerManager.cs]: shouldnt be in servermanager

File: src/NosCore.Controllers/CharacterScreenPacketController.cs [hunk 18]
```diff
@@ -315,6 +317,14 @@ public void SelectCharacter(SelectPacket selectPacket)
                     Session.Character.Mp = (int) Session.Character.MPLoad();
                 }
 
+                var characterRelationPartitioner = Partitioner.Create(DAOFactory.CharacterRelationDAO.Where(s => s.CharacterId == Session.Character.CharacterId || s.RelatedCharacterId == Session.Character.CharacterId).Cast<CharacterRelation>(), EnumerablePartitionerOptions.NoBuffering);
+
+                Parallel.ForEach(characterRelationPartitioner, relation =>
+                {
+                    relation.CharacterName = DAOFactory.CharacterDAO.FirstOrDefault(s => s.CharacterId == relation.RelatedCharacterId)?.Name;
```
Comment (0Lucifer0) [inline src/NosCore.Controllers/CharacterScreenPacketController.cs]: no you need to use a where to only query once the database

File: src/NosCore.Controllers/CharacterScreenPacketController.cs [hunk 19]
```diff
@@ -315,6 +318,14 @@ public void SelectCharacter(SelectPacket selectPacket)
                     Session.Character.Mp = (int) Session.Character.MPLoad();
                 }
 
+                IEnumerable<CharacterRelation> relations = DAOFactory.CharacterRelationDAO.Where(s => s.CharacterId == Session.Character.CharacterId).Cast<CharacterRelation>();
+
+                foreach (CharacterRelation relation in relations)
+                {
+                    relation.CharacterName = DAOFactory.CharacterDAO.Where(s => s.CharacterId == relation.RelatedCharacterId)?.FirstOrDefault()?.Name;
```
Comment (0Lucifer0) [inline src/NosCore.Controllers/CharacterScreenPacketController.cs]: still need to be outside of the foreach, just use a where(in list) and iterate on this saved list

File: src/NosCore.DAL/DAOFactory.cs [hunk 20]
```diff
@@ -43,6 +43,7 @@ public static class DAOFactory
         private static GenericDAO<Skill, SkillDTO> _skillDAO;
         private static GenericDAO<NpcMonsterSkill, NpcMonsterSkillDTO> _npcMonsterSkillDAO;
         private static GenericDAO<MapMonster, MapMonsterDTO> _mapMonsterDAO;
+        private static GenericDAO<Database.Entities.CharacterRelation, CharacterRelationDTO> _characterRelationDAO;
```
Comment (0Lucifer0) [inline src/NosCore.DAL/DAOFactory.cs]: no Database.Entities

File: src/NosCore.DAL/DAOFactory.cs [hunk 21]
```diff
@@ -142,6 +143,9 @@ public static class DAOFactory
         public static GenericDAO<MapMonster, MapMonsterDTO> MapMonsterDAO =>
             _mapMonsterDAO ?? (_mapMonsterDAO = new GenericDAO<MapMonster, MapMonsterDTO>(_mapper));
 
+        public static GenericDAO<Database.Entities.CharacterRelation, CharacterRelationDTO> CharacterRelationDAO =>
```
Comment (0Lucifer0) [inline src/NosCore.DAL/DAOFactory.cs]: ditto

File: src/NosCore.DAL/DAOFactory.cs [hunk 22]
```diff
@@ -142,6 +143,9 @@ public static class DAOFactory
...
+            _characterRelationDAO ?? (_characterRelationDAO = new GenericDAO<Database.Entities.CharacterRelation, CharacterRelationDTO>(_mapper));
```
Comment (0Lucifer0) [inline src/NosCore.DAL/DAOFactory.cs]: ditto

File: src/NosCore.Data/WebApi/ConnectedCharacter.cs [hunk 23]
```diff
@@ -1,7 +1,13 @@
...
+
+namespace NosCore.Data.WebApi
 {
     public class ConnectedCharacter
     {
         public string Name { get; set; }
+
+        public long Id { get; set; }
+
+        public ICollection<CharacterRelation> Relations { get; set; }
```
Comment (0Lucifer0) [inline src/NosCore.Data/WebApi/ConnectedCharacter.cs]: need to be in another object with another controller

File: src/NosCore.GameObject/Character.cs [hunk 24]
```diff
@@ -162,6 +189,153 @@ public int IsReputHero()
...
+            foreach (CharacterRelation relation in CharacterRelations.Values)
+            {
+                var target = ServerManager.Instance.Sessions.Values.FirstOrDefault(s => s.Character.CharacterId == relation.RelatedCharacterId);
+                if (target != null)
+                {
+                    target.SendPacket(target.Character.GenerateFinit());
+                }
+                else
+                {
+                    foreach (var server in WebApiAccess.Instance.Get<List<WorldServerInfo>>("api/channels"))
```
Comment (0Lucifer0) [inline src/NosCore.GameObject/Character.cs]: dont use this in a foreach

File: src/NosCore.GameObject/Character.cs [hunk 25]
```diff
@@ -162,6 +189,153 @@ public int IsReputHero()
...
+                    {
+                        var accounts = WebApiAccess.Instance.Get<List<ConnectedAccount>>("api/connectedAccounts", server.WebApi);
```
Comment (0Lucifer0) [inline src/NosCore.GameObject/Character.cs]: dito

File: src/NosCore.GameObject/Character.cs [hunk 26]
```diff
@@ -162,6 +189,153 @@ public int IsReputHero()
...
+            {
+                return;
+            }
+
+            DAOFactory.CharacterRelationDAO.Delete(targetRelation.CharacterRelationId);
+            CharacterRelations.TryRemove(characterRelation.CharacterRelationId, out _);
+            Session.SendPacket(GenerateFinit());
+            //TODO: Refresh target relation if online, this is just for testing purposes
+            ClientSession target = ServerManager.Instance.Sessions.Values.FirstOrDefault(s => s.Character.CharacterId == characterId);
+            target?.SendPacket(target?.Character.GenerateFinit());
```
Comment (0Lucifer0) [inline src/NosCore.GameObject/Character.cs]: you also need to remove the relation from this session. Moreover this step need to be before database deletion

File: src/NosCore.GameObject/Character.cs [hunk 27]
```diff
@@ -162,6 +189,153 @@ public int IsReputHero()
...
+        }
+
+        public bool IsBlockedByCharacter(long characterId)
+        {
+            return CharacterRelations.Values.Any(s => s.RelationType == CharacterRelationType.Blocked && s.RelatedCharacterId.Equals(characterId) && !s.RelatedCharacterId.Equals(CharacterId) && s.CharacterId.Equals(CharacterId));
+        }
+
+        public bool IsFriendOfCharacter(long characterId)
+        {
+            return CharacterRelations.Values.Any(s => s.RelationType == CharacterRelationType.Friend && s.RelatedCharacterId.Equals(characterId) && s.CharacterId.Equals(CharacterId));
```
Comment (0Lucifer0) [inline src/NosCore.GameObject/Character.cs]: can be merged into IsRelatedToCharacter with a RelationType switch

File: src/NosCore.GameObject/Networking/BroadcastableBase.cs [hunk 28]
```diff
@@ -58,6 +58,19 @@ private void SpreadBroadcastpacket(BroadcastPacket sentPacket)
 
             switch (sentPacket.Receiver)
             {
+                case ReceiverType.AllExceptMeAndBlacklisted:
+                    Parallel.ForEach(
+                        Sessions.Where(s => s.Value.Character.CharacterId != sentPacket.Sender.Character.CharacterId && !s.Value.Character.IsBlockedByCharacter(sentPacket.Sender.Character.CharacterId)),
+                        session =>
+                        {
+                            if (!session.Value.HasSelectedCharacter)
```
Comment (0Lucifer0) [inline src/NosCore.GameObject/Networking/BroadcastableBase.cs]: add this clause into the where

File: src/NosCore.Controllers/CharacterScreenPacketController.cs [hunk 29]
```diff
@@ -315,6 +318,15 @@ public void SelectCharacter(SelectPacket selectPacket)
                     Session.Character.Mp = (int) Session.Character.MPLoad();
                 }
 
+                IEnumerable<CharacterRelationDTO> relations = DAOFactory.CharacterRelationDAO.Where(s => s.CharacterId == Session.Character.CharacterId);
+                List<CharacterDTO> characters = DAOFactory.CharacterDAO.Where(s => relations.Select(v => v.RelatedCharacterId).Contains(s.CharacterId)).ToList();
```
Comment (erwan-joly) [inline src/NosCore.Controllers/CharacterScreenPacketController.cs]: can you manually add something in db and check this doesnt crash ?

Comment (Elendan) [reply src/NosCore.Controllers/CharacterScreenPacketController.cs]: I've tested it, it works fine

File: src/NosCore.Data/WebApi/ConnectedCharacter.cs [hunk 30]
```diff
@@ -1,7 +1,13 @@
...
+
+namespace NosCore.Data.WebApi
 {
     public class ConnectedCharacter
     {
         public string Name { get; set; }
+
+        public long Id { get; set; }
+
+        public ICollection<CharacterRelationDTO> Relations { get; set; }
```
Comment (erwan-joly) [inline src/NosCore.Data/WebApi/ConnectedCharacter.cs]: has no reason to be there relation need a special controller

File: src/NosCore.Data/WebApi/RelationData.cs [hunk 31]
```diff
@@ -0,0 +1,13 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace NosCore.Data.WebApi
+{
+    public class RelationData
+    {
+        public long RelationId { get; set; }
+
+        public bool IsConnected { get; set; }
```
Comment (erwan-joly) [inline src/NosCore.Data/WebApi/RelationData.cs]: a relation shouldnt know if it is connected

File: src/NosCore.GameObject/Character.cs [hunk 32]
```diff
@@ -163,6 +190,228 @@ public int IsReputHero()
             return 0;
         }
 
+        public void UpdateFriendList(bool isConnected)
```
Comment (erwan-joly) [inline src/NosCore.GameObject/Character.cs]: shoudlnt have connected here

File: src/NosCore.WorldServer/Controllers/RelationsController.cs [hunk 33]
```diff
@@ -0,0 +1,57 @@
...
+
+namespace NosCore.WorldServer.Controllers
+{
+    [Route("api/[controller]")]
+    [AuthorizeRole(AuthorityType.GameMaster)]
+    public class RelationsController : Controller
+    {
+        // POST api/relations
+        [HttpPost]
+        public IActionResult PostUpdateFriendList([FromBody] PostedPacket postedPacket)
```
Comment (erwan-joly) [inline src/NosCore.WorldServer/Controllers/RelationsController.cs]: this is not a post you are trying to post a packet it has nothing to do here use packet api

File: src/NosCore.WorldServer/Controllers/RelationsController.cs [hunk 34]
```diff
@@ -0,0 +1,57 @@
...
+
+            var session = ServerManager.Instance.Sessions.Values.FirstOrDefault(s => s.Character.CharacterId == postedPacket.ReceiverCharacterData.CharacterId);
+
+            session?.SendPacket(session.Character.GenerateFinfo(postedPacket.SenderCharacterData.CharacterId, postedPacket.SenderCharacterData.RelationData.IsConnected ));
+
+            return Ok();
+        }
+
+        // POST api/relations/deleteRelation
+        [HttpPost("deleteRelation")]
```
Comment (erwan-joly) [inline src/NosCore.WorldServer/Controllers/RelationsController.cs]: no Post but delete, no new route

File: src/NosCore.WorldServer/Controllers/RelationsController.cs [hunk 35]
```diff
@@ -0,0 +1,57 @@
...
+
+            var session = ServerManager.Instance.Sessions.Values.FirstOrDefault(s => s.Character.CharacterId == postedPacket.ReceiverCharacterData.CharacterId);
+
+            if (session == null)
+            {
+                return Ok();
+            }
+
+            session.Character.CharacterRelations.TryRemove(postedPacket.ReceiverCharacterData.RelationData.RelationId, out _);
+            session.SendPacket(session.Character.GenerateFinit());
```
Comment (erwan-joly) [inline src/NosCore.WorldServer/Controllers/RelationsController.cs]: not here

File: src/NosCore.WorldServer/Controllers/RelationsController.cs [hunk 36]
```diff
@@ -0,0 +1,57 @@
...
+            var session = ServerManager.Instance.Sessions.Values.FirstOrDefault(s => s.Character.CharacterId == postedPacket.ReceiverCharacterData.CharacterId);
+
+            session?.SendPacket(session.Character.GenerateFinfo(postedPacket.SenderCharacterData.CharacterId, postedPacket.SenderCharacterData.RelationData.IsConnected ));
+
+            return Ok();
+        }
+
+        // POST api/relations/deleteRelation
+        [HttpPost("deleteRelation")]
+        public IActionResult DeleteRelation([FromBody] PostedPacket postedPacket)
```
Comment (erwan-joly) [inline src/NosCore.WorldServer/Controllers/RelationsController.cs]: PostedPacket shouldnt be the object you have to pass an ID

File: src/NosCore.WorldServer/Controllers/connectedAccountsController.cs [hunk 37]
```diff
@@ -18,13 +18,13 @@ public class ConnectedAccountsController : Controller
...
         {
-            return ServerManager.Instance.Sessions.Select(s =>
+            return ServerManager.Instance.Sessions.Values.Select(s =>
                 new ConnectedAccount
                 {
                     Name = s.Account.Name,
                     Language = s.Account.Language,
                     ChannelId = MasterClientListSingleton.Instance.ChannelId,
-                    ConnectedCharacter = s.Character == null ? null : new ConnectedCharacter() { Name = s.Character.Name }
+                    ConnectedCharacter = s.Character == null ? null : new ConnectedCharacter() { Name = s.Character.Name, Id = s.Character.CharacterId, Relations = s.Character.CharacterRelations.Values },
```
Comment (erwan-joly) [inline src/NosCore.WorldServer/Controllers/connectedAccountsController.cs]: Relation shouldnt be here

Comment (erwan-joly) [inline src/NosCore.WorldServer/Controllers/connectedAccountsController.cs]: Character should maybe not even be here

File: test/NosCore.Tests/PacketGenerationTests.cs [hunk 38]
```diff
@@ -45,6 +70,30 @@ public void GenerateInPacketIsNotCorruptedForMonster()
...
+            var subpacket = new List<NsTeStSubPacket>();
+            subpacket.Add(new NsTeStSubPacket
+            {
+                Host = "-1",
+                Port = null,
+                Color = null,
+                WorldCount = 10000,
+                WorldId = 10000,
+                Name = "1"
+            }); //useless server to end the client reception
```
Comment (erwan-joly) [inline test/NosCore.Tests/PacketGenerationTests.cs]: remove the comment please copy pasta error mybad

File: src/NosCore.Controllers/CharacterScreenPacketController.cs (hunk 29 above)
Comment (erwan-joly) [reply src/NosCore.Controllers/CharacterScreenPacketController.cs]: was just to make sure linq to sql was enable to handle it

**Pull request merged**